### PR TITLE
Remove default gender from Racers

### PR DESCRIPTION
**Pull request by moveson**
We had a couple of instances of women entering themselves as men because the gender is set to Male by default. 

This PR proposes to remove the default and require an entrant to pick a gender from the dropdown menu, reducing the chances of this problem happening in the future.

### CODE REVIEW

File: db/migrate/20200913140801_remove_default_from_racer_gender.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RemoveDefaultFromRacerGender < ActiveRecord::Migration[5.1]
+  def change
+    change_column_default :racers, :gender, from: 0, to: nil
```
Comment (moveson) [inline db/migrate/20200913140801_remove_default_from_racer_gender.rb:3]: The `from:` and `to:` syntax makes this migration reversible.

File: app/views/race_editions/enter.html.erb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
     <%= race_edition_form.fields_for :racers, @racer do |racer_fields| %>
       <%= racer_fields.label :first_name %>
-      <%= racer_fields.text_field :first_name %>
+      <%= racer_fields.text_field :first_name, autofocus: true %>
```
Comment (moveson) [inline app/views/race_editions/enter.html.erb:9]: Not strictly related to this PR, but I think this change marginally improves the user experience.

File: app/views/race_editions/enter.html.erb [hunk 3]
```diff
@@ -15,7 +15,7 @@
       <%= racer_fields.text_field :email %>
     
       <%= racer_fields.label :gender %>
-      <%= racer_fields.select :gender, Racer.genders.keys, {} %>
+      <%= racer_fields.select :gender, Racer.genders.keys, include_blank: true %>
```
Comment (moveson) [inline app/views/race_editions/enter.html.erb:18]: No gender will be auto-selected; entrants will have to affirmatively pick a gender. If the form is submitted without a gender chosen, validation will fail and the form will be re-rendered.

File: db/schema.rb [hunk 4]
```diff
@@ -59,7 +59,7 @@
   create_table "racers", force: :cascade do |t|
     t.string "first_name", null: false
     t.string "last_name", null: false
-    t.integer "gender", default: 0
+    t.integer "gender"
```
Comment (moveson) [inline db/schema.rb:62]: If we don't make this change, then "male" will automatically be selected on the gender dropdown menu in the form.

File: app/views/race_editions/enter.html.erb (hunk 2 above)
Comment (billwright) [reply app/views/race_editions/enter.html.erb:9]: Love it.

File: db/migrate/20200913140801_remove_default_from_racer_gender.rb (hunk 1 above)
Comment (billwright) [reply db/migrate/20200913140801_remove_default_from_racer_gender.rb:3]: Great call on this PR.

File: db/schema.rb (hunk 4 above)
Comment (billwright) [reply db/schema.rb:62]: Agreed

**Pull request merged**